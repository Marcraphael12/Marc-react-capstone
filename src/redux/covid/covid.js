import { fetchFromApi, catchFromApi } from '../../apis/countryData';

const LOAD_DATA = 'covid-data-app/covid/LOAD_DATA';
const SELECT_DATA = 'covid-data-app/covid/SELECT_DATA';
const HISTORY_DATA = 'covid-data-app/covid/HISTORY_DATA';
const DATA_LOADING = 'covid-data-app/covid/DATA_LOADING';

const initialState = {
  countries: {},
};

const covidReducer = (state = initialState, action) => {
  switch (action.type) {
    case LOAD_DATA:
      return {
        ...state,
        countries: action.payload.countries,
        total: action.payload.total,
        loading: false,
      };
    case SELECT_DATA:
      return {
        ...state,
        currentCountry: action.payload,
        loading: false,
      };

    case HISTORY_DATA:
      return {
        ...state,
        currentHistory: action.payload,
        loading: false,
      };
    case DATA_LOADING:
      return {
        ...state,
        loading: true,
      };
    default:
      return state;
  }
};

const loadData = (payload) => ({
  type: LOAD_DATA,
  payload,
});

const selectData = (payload) => ({
  type: SELECT_DATA,
  payload,
});

const loadHistoryData = (payload) => ({
  type: HISTORY_DATA,
  payload,
});
const dataLoading = (payload) => ({
  type: DATA_LOADING,
  payload,
});

const loadDataThunk = () => async (dispatch) => {
  try {
    const data = await fetchFromApi();
    dispatch(loadData(data));
  } catch (e) {
    const data = await catchFromApi();
    dispatch(loadData(data));
  }
};

const loadHistoryThunk = (country) => async (dispatch) => {
  const data = await fetchFromApi(country);
  const { dates } = data;
  if (dates) {
    dispatch(loadHistoryData(dates));
  }
};

export {
  covidReducer, loadHistoryThunk, loadDataThunk, loadData, selectData, dataLoading,
};