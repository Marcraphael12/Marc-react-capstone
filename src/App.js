import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Routes, Route } from 'react-router-dom';
import Navbar from './components/Nav';
import Countries from './components/countries/Countries';
import CountryInfo from './components/countries/country_infos';
import { dataLoading, loadDataThunk, selectData } from './Redux/covid/covid';
import countryNames from './components/countries/countries_name';
import Footer from './components/Footer';
import './scss/app.scss';

function App() {
  const countries = useSelector((state) => state.covidReducer.countries);
  const total = useSelector((state) => state.covidReducer.total);
  const load = useSelector((state) => state.covidReducer.loading);
  const currentCntry = useSelector((state) => state.covidReducer.currentCountry);
  const isoName = countryNames[currentCntry];

  const mapImg = isoName ? `https://raw.githubusercontent.com/djaiss/mapsicon/33ba28808f8d32b5bae0ffada9cadd07073852e1/all/${isoName.toLowerCase()}/vector.svg` : '/World_map.png';
  const dispatch = useDispatch();

  const handleClick = (country) => dispatch(selectData(country));
  useEffect(() => {
    dispatch(dataLoading());
    dispatch(loadDataThunk());
  }, [dispatch]);
  return (
    <div className="app">
      <Navbar />
      <Routes>
        <Route exact path="/" element={<Countries countries={countries} total={total} loading={load} handleClick={handleClick} />} />
        <Route path="/country" element={<CountryInfo current={currentCntry} image={mapImg} />} />
      </Routes>
      <Footer />
    </div>
  );
}

export default App;;
