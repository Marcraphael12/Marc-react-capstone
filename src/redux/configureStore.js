import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import { covidReducer } from './covid/covid';

const reducer = combineReducers({
  covidReducer,
});

const middleware = [logger, thunk];

const store = createStore(
  reducer,
  applyMiddleware(...middleware),
);

export default store;
