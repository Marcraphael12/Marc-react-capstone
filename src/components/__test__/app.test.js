import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import { createStore, combineReducers, applyMiddleware } from 'redux';
import App from '../../App';
import { covidReducer } from '../../Redux/covid/covid';

const reducer = combineReducers({
  covidReducer,

});

const store = createStore(
  reducer,
  applyMiddleware(logger, thunk),
);

describe('App', () => {
  it('renders ! ', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>,
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});