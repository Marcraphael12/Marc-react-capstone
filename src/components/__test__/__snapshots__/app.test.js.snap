// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders !  1`] = `
<div
  className="app"
>
  <nav>
    <div
      className="location-path"
    >
      <svg
        className="icon-virus"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M483.55,227.55H462c-50.68,0-76.07-61.27-40.23-97.11L437,115.19A28.44,28.44,0,0,0,396.8,75L381.56,90.22c-35.84,35.83-97.11,10.45-97.11-40.23V28.44a28.45,28.45,0,0,0-56.9,0V50c0,50.68-61.27,76.06-97.11,40.23L115.2,75A28.44,28.44,0,0,0,75,115.19l15.25,15.25c35.84,35.84,10.45,97.11-40.23,97.11H28.45a28.45,28.45,0,1,0,0,56.89H50c50.68,0,76.07,61.28,40.23,97.12L75,396.8A28.45,28.45,0,0,0,115.2,437l15.24-15.25c35.84-35.84,97.11-10.45,97.11,40.23v21.54a28.45,28.45,0,0,0,56.9,0V462c0-50.68,61.27-76.07,97.11-40.23L396.8,437A28.45,28.45,0,0,0,437,396.8l-15.25-15.24c-35.84-35.84-10.45-97.12,40.23-97.12h21.54a28.45,28.45,0,1,0,0-56.89ZM224,272a48,48,0,1,1,48-48A48,48,0,0,1,224,272Zm80,56a24,24,0,1,1,24-24A24,24,0,0,1,304,328Z"
        />
      </svg>
      <h1
        className="back"
      >
        Covid-19 world statistics
      </h1>
    </div>
  </nav>
  <div
    className="header-container"
  >
    <div
      className="header-wrapper"
    >
      <div
        className="world-info"
      >
        <span
          className="world-cases"
        >
          Roughly 
        </span>
        <span
          className="world-cases-value"
        >
           
          NaN
        </span>
        <span
          className="world-cases"
        >
          Covid world Cases 
        </span>
      </div>
      <form
        className="form"
      >
        <input
          className="input-search"
          onChange={[Function]}
          placeholder="Search a country"
          type="text"
        />
      </form>
    </div>
    <ul
      className="countries-list"
    />
  </div>
  <footer
    className="footer"
  >
    <div
      className="footer-container"
    >
      <h3
        className="authorr"
      >
        Â© Marc Raphael 
      </h3>
      <div
        className="icons"
      >
        <a
          href="/https://github.com/Marcraphael12"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            className="git fab fa-github-square"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM727.3 401.7c.3 4.7.3 9.6.3 14.4 0 146.8-111.8 315.9-316.1 315.9-63 0-121.4-18.3-170.6-49.8 9 1 17.6 1.4 26.8 1.4 52 0 99.8-17.6 137.9-47.4-48.8-1-89.8-33-103.8-77 17.1 2.5 32.5 2.5 50.1-2a111 111 0 0 1-88.9-109v-1.4c14.7 8.3 32 13.4 50.1 14.1a111.13 111.13 0 0 1-49.5-92.4c0-20.7 5.4-39.6 15.1-56a315.28 315.28 0 0 0 229 116.1C492 353.1 548.4 292 616.2 292c32 0 60.8 13.4 81.1 35 25.1-4.7 49.1-14.1 70.5-26.7-8.3 25.7-25.7 47.4-48.8 61.1 22.4-2.4 44-8.6 64-17.3-15.1 22.2-34 41.9-55.7 57.6z"
            />
          </svg>
        </a>
        <a
          href="/https://twitter.com/MarcRaphael20"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            className="twitter fab fa-twitter-square"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
            />
          </svg>
        </a>
        <a
          href="/https://www.linkedin.com/in/essogo-raphael/"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            className="linkedin fab fa-linkedin"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM349.3 793.7H230.6V411.9h118.7v381.8zm-59.3-434a68.8 68.8 0 1 1 68.8-68.8c-.1 38-30.9 68.8-68.8 68.8zm503.7 434H675.1V608c0-44.3-.8-101.2-61.7-101.2-61.7 0-71.2 48.2-71.2 98v188.9H423.7V411.9h113.8v52.2h1.6c15.8-30 54.5-61.7 112.3-61.7 120.2 0 142.3 79.1 142.3 181.9v209.4z"
            />
          </svg>
        </a>
      </div>
    </div>
  </footer>
</div>
`;
